### PR TITLE
Suggested improvement to font_init 

### PR DESCRIPTION
**Pull request by PurityLake**
Refers to #3394 

This is a suggested solution to the Font class. It simply makes the font name and size arguments options with default values of "freesansbold.ttf" and 12 respectively.

This wouldn't be a breaking change as the default usual behavior still works, it just made the args optional instead of mandatory.

This is being marked as a draft so it can be discussed.

### CODE REVIEW

File: src_c/font.c [hunk 1]
```diff
@@ -695,15 +697,14 @@ font_dealloc(PyFontObject *self)
...
-    PyObject *obj;
+    PyObject *obj = Py_None;
     SDL_RWops *rw;
 
     self->font = NULL;
-    if (!PyArg_ParseTuple(args, "Oi", &obj, &fontsize)) {
-        return -1;
-    }
+    // This is now optional so no need to return if it doesn't pass
+    PyArg_ParseTuple(args, "|Oi", &obj, &fontsize);
```
Comment (ankith26) [inline src_c/font.c:707]: The `return -1;` part is still needed for if `PyArg_ParseTuple` errors

**Comment by ankith26** [comment]
Hmm, while we are at the topic of updates to this function signature, how about adding kwargs support in here as well? (as comments in the linked issue suggest as well)

To do that you must use [`PyArg_ParseTupleAndKeywords`](https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTupleAndKeywords) instead of `PyArg_ParseTuple`

**Comment by PurityLake** [comment]
Also updated the Font docs

File: docs/reST/ref/font.rst [hunk 2]
```diff
@@ -167,7 +168,8 @@ solves no longer exists, it will likely be removed in the future.
    the height of the font in pixels. If the filename is ``None`` the pygame
    default font will be loaded. If a font cannot be loaded from the arguments
    given an exception will be raised. Once the font is created the size cannot
-   be changed.
+   be changed. If no arguments are given then the default font will be used and
+   a font size of 32 is used.
```
Comment (ankith26) [inline docs/reST/ref/font.rst:172]: The docs say the default size is 32, but 12 is implemented in the code

**Review by ankith26 — COMMENTED** [review]
Since this is new API, this PR is being put off to 2.1.4, so that more people can get to reviewing it and providing more inputs.

Also as new API, a few updates to the unit tests to test for this is also needed

**Review by MyreMylar — CHANGES_REQUESTED** [review]
All seems fine to me, though I'd like a mini debate on a good name for a keyword argument that represents a python file like object. 

Which is why I'm leaving this on request changes.

File: src_c/doc/font_doc.h [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #define DOC_PYGAMEFONTGETFONTS "get_fonts() -> list of strings\nget all available fonts"
 #define DOC_PYGAMEFONTMATCHFONT "match_font(name, bold=False, italic=False) -> path\nfind a specific font on the system"
 #define DOC_PYGAMEFONTSYSFONT "SysFont(name, size, bold=False, italic=False) -> Font\ncreate a Font object from the system fonts"
-#define DOC_PYGAMEFONTFONT "Font(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
+#define DOC_PYGAMEFONTFONT "Font(file=None, size=12) -> Font\nFont(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
```
[the remark this answers is not in the document]
Comment (PurityLake) [reply src_c/doc/font_doc.h:11]: Hmmm, yes file is wrong. I just checked the docs and I realised I misread it as file, not filename. Unless we want to support being passed a file-like object I'd be happy to change it but maybe that's too much of a change? But we also support pathlib.Path so maybe a different name like "filepath" might be better? It suggests we can pass it as a string or Path object

[the remark this answers is not in the document]
Comment (MyreMylar) [reply src_c/doc/font_doc.h:11]: Underneath in the C code it already does accept a python file like object as well as a file path via SDL RWops. 

I'm just trying to come up with a single name for the keyword argument that encapsulates all these potential inputs for the first parameter. In the past it didn't matter as we were exclusively passing by value so the first parameter could be named differently in the docs depending on the input, but once you turn it into a python keyword you need to use the same keyword when passing in a file path (in string, or Path object format) or a file object.

Does that make sense?

I think I'm currently preferring `source` as a universal parameter name, but `file_path` has the benefit of being by far the most frequently used type of input for the first parameter.

[the remark this answers is not in the document]
Comment (PurityLake) [reply src_c/doc/font_doc.h:11]: I realise my "new suggestion" was the exact same as I had it lol

I'm thinking source or resource. Either one I'd be happy with.

[the remark this answers is not in the document]
Comment (PurityLake) [reply src_c/doc/font_doc.h]: I have made that change now

**Comment by PurityLake** [comment]
No idea why this isn't passing, in any of the instances, the tests go fine for me on Ubuntu 22.04

**Comment by PurityLake** [comment]
Passes now thankfully

**Review by illume — APPROVED** [review]
👍 thanks

**Pull request merged**